.github/workflows/release.yml

```diff
@@ -59,3 +65,78 @@ jobs:
     - name: Prune docker layers cache
       shell: bash
       run: just docker-cache-prune
+
+  cni_validator_build:
+    runs-on: ubuntu-20.04
+    needs: [tag]
+    timeout-minutes: 30
+    permissions:
+      id-token: write # needed for signing the images with GitHub OIDC token
+    strategy:
+      matrix:
+        arch:
+        - amd64
+        - arm
+        - arm64
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+    - name: Set tag
+      run: echo 'TAG=${{ needs.tag.outputs.tag }}' >> "$GITHUB_ENV"
+    - uses: extractions/setup-just@aa5d15c144db4585980a44ebfdd2cf337c4f14cb
+    - uses: docker/setup-buildx-action@dc7b9719a96d48369863986a06765841d7ea23f6
+    - uses: actions/cache@0865c47f36e68161719c5b124609996bb5c40129
+      with:
+        path: ${{ env.DOCKER_BUILDKIT_CACHE }}
+        key: cni-validator-${{ matrix.arch }}-${{ env.TAG }}
+        restore-keys: cni-validator-
+    - uses: docker/login-action@49ed152c8eca782a232dede0303416e8f356c37b
+      with:
+        registry: ghcr.io
+        username: ${{ secrets.DOCKER_GHCR_USERNAME }}
+        password: ${{ secrets.DOCKER_GHCR_PAT }}
+    - name: Build docker images
+      shell: bash
+      run: |
+          docker buildx build . \
+          --file="cni-validator/${{ matrix.arch }}.dockerfile" \
+          --tag="ghcr.io/${{ github.repository_owner }}/cni-validator:${TAG}-${{ matrix.arch }}" \
+          --cache-from type=local,src="${DOCKER_BUILDKIT_CACHE}" \
+          --cache-to type=local,dest="${DOCKER_BUILDKIT_CACHE}",mode=max \
+          --platform="${{ matrix.arch }}" \
+          --push
```

Comment:
I have a feeling we can simplify this because we're not taking dependencies like rustls or openssl. We can use musl to build static binaries... This means we may even be able to do normal docker multiarch builds without the special arch-specific tags, etc.

Reply:
See, for example: https://github.com/olix0r/tcp-echo/blob/9e464a54e71ef6a3d5430875997eaed7048e67db/Dockerfile

Reply:
Happy to simplify this. Curious though, we generally opt to use glibc (GNU) to dynamically link against libs like openssl to get security updates? Is there any other advantage? What makes musl builds harder to use with deps like rustls or openssl?